### PR TITLE
Move npm related state tests to pytest

### PR DESCRIPTION
**Pull request by krionbsd**
### What does this PR do?
Move npm related state tests to pytest

### CODE REVIEW

**Review by twangboy — DISMISSED** [review]

File: tests/pytests/unit/states/test_npm.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    mock_err = MagicMock(side_effect=CommandExecutionError)
+    mock_dict = MagicMock(return_value={name: {"version": "1.2"}})
+    with patch.dict(npm.__salt__, {"npm.list": mock_err}):
+        comt = "Error looking up 'coffee-script': "
+        ret.update({"comment": comt})
+        assert npm.installed(name) == ret
+
+    with patch.dict(npm.__salt__, {"npm.list": mock_dict, "npm.install": mock_err}):
+        with patch.dict(npm.__opts__, {"test": True}):
+            comt = "Package(s) 'coffee-script' " "satisfied by coffee-script@1.2"
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_npm.py:34]:
```suggestion
            comt = "Package(s) 'coffee-script' satisfied by coffee-script@1.2"
```

File: tests/pytests/unit/states/test_npm.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            ret.update({"comment": comt, "result": True})
+            assert npm.installed(name) == ret
+
+        with patch.dict(npm.__opts__, {"test": False}):
+            comt = "Package(s) 'coffee-script' " "satisfied by coffee-script@1.2"
```
Comment (s0undt3ch) [inline tests/pytests/unit/states/test_npm.py:39]:
```suggestion
            comt = "Package(s) 'coffee-script' satisfied by coffee-script@1.2"
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/states/test_npm.py (hunk 1 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_npm.py]: Fixed

File: tests/pytests/unit/states/test_npm.py (hunk 2 above)
Comment (krionbsd) [reply tests/pytests/unit/states/test_npm.py]: Fixed

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**